### ISSUE TITLE
ArgumentNullException "rateManager" when application is quitting

### ISSUE THREAD

**Issue by renanwolf** [renanwolf/Unity-LowPowerMemoryConsumption#2]
Some times, when the application is quitting, depending on the use case, an ArgumentNullException is thrown when Unity has already destroyed the RateManager GameObject.

```
ArgumentNullException: Value cannot be null. Parameter name: rateManager

UniRate.RateRequest..ctor (UniRate.RateManager rateManager) (at <00000000000000000000000000000000>:0)
UniRate.RateManager.RequestUpdateRate (System.Int32 updateRate) (at <00000000000000000000000000000000>:0)
*****.SetUpdateRateRequest (System.Boolean enabled) (at <00000000000000000000000000000000>:0)
System.Action`1[T].Invoke (T obj) (at <00000000000000000000000000000000>:0)
*****PropertyRx`1[TValue].OnValueChanged (TValue fromValue, TValue toValue) (at <00000000000000000000000000000000>:0)
*****.ReleaseItems () (at <00000000000000000000000000000000>:0)
*****ComponentBase.OnDisable () (at <00000000000000000000000000000000>:0)
*****LoaderComponent`1[TItem].OnDisable () (at <00000000000000000000000000000000>:0)
```

**Comment by renanwolf** [comment]
As we can see on the stack trace, it begins at the disabling of Unity components. Since the `OnApplicationQuit()` is invoked before `OnDisable()` (accordingly to [Unity docs](https://docs.unity3d.com/Manual/ExecutionOrder.html)), it might be a good idea to implement a private flag `_isApplicationQuitting` inside the `RateManager` and prevent any type of `RateRequest` to be created if the flag is true. Also log (maybe warning) something like "not creating request because application is quitting".

**Comment by renanwolf** [comment]
Looks like it was fixed in  ff8273ad30f78285656e7ca3fb4a2392eab4f682
